### PR TITLE
Prometheus: Add browser access mode deprecation warning

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
As a part of data source backend migration for Prometheus data source, we are adding browser access mode deprecation warning. This is because when the data source is fully migrated, browser mode won't be supported anymore. Influx and Elastic already have this deprecation warning.

More https://docs.google.com/document/d/1CzmOgShzXmtac9Tcq5CT8-4_q3n1i2Kd88nZDEZCLGw

### CODE REVIEW

File: docs/sources/datasources/prometheus.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ To access Prometheus settings, hover your mouse over the **Configuration** (gear
...
 | `Default`                 | Default data source that is pre-selected for new panels.                                                                                                                                                                                                          |
 | `Url`                     | The URL of your Prometheus server, for example, `http://prometheus.example.org:9090`.                                                                                                                                                                             |
-| `Access`                  | Server (default) = URL needs to be accessible from the Grafana backend/server, Browser = URL needs to be accessible from the browser.                                                                                                                             |
+| `Access`                  | Server (default) = URL needs to be accessible from the Grafana backend/server, Browser = URL needs to be accessible from the browser. **Note**: Browser access is deprecated and will be removed in a future release.                                             |
```
Comment (ivanahuckova) [inline docs/sources/datasources/prometheus.md:22]: This is the change, everything else is just the formatting update.

**Review by gabor — APPROVED** [review]
LGTM!

**Review by Elfo404 — APPROVED** [review]

**Comment by zecke** [comment]
I am late to this. Where can I learn more about "data source backend"? Do you have a timeline for the removal?

**Comment by gabor** [comment]
@grafana/observability-metrics should have the answer 👍

**Pull request merged**